Gopkg.lock

```diff
@@ -828,6 +828,38 @@
   pruneopts = ""
   revision = "2fe55d9e1d0704d3c6f03f69a1fd9ebe2aef9df1"
 
+[[projects]]
+  branch = "master"
+  digest = "1:0c210be2169af89f30c3b260861bae766b79b8e8eef46530670e25f755cd853b"
+  name = "github.com/mdlayher/genetlink"
+  packages = ["."]
+  pruneopts = ""
+  revision = "e35f2bf499b9e396bf88e96af81ac1085f5c6f0b"
+
+[[projects]]
+  branch = "master"
+  digest = "1:d9ee5e12190ff796cadfad8a9aae2ad57a303b1d4d476518ace3a5dc01ae4c3a"
+  name = "github.com/mdlayher/netlink"
```

Comment:
This is a transitive dependency introduced by the addition of the direct dependency github.com/mdlayher/wireguardctrl. Do you mean you would prefer re-implementing a portion of github.com/mdlayher/wireguardctrl using the existing netlink library, and avoid introducing additional dependencies entirely? Or do you mean you would prefer using a fork of github.com/mdlayher/wireguardctrl that uses vishvananda/netlink so as to not introduce a new transitive netlink dependency?

Reply:
(I would probably prefer the latter, since I'm planning on forking mdlayher/wireguardctrl anyways for use in a separate future project, but happy to try whichever approach unblocks merging here)

Reply:
The most ideal would be the former, since it would result in the smallest overall code size, but the either way would be acceptable.

It looks like there are 2 methods for interfacing with wireguard: netlink or a userspace unix socket/named pipe.  On Linux, is the unix socket created even when using the wireguard interface, or is it only created when using the userspace implementation?  If the socket was always created that could be the ideal way for us to monitor it.

Reply:
A Wireguard Unix socket is only created in userspace implementations (and even then it is just an I/O convention, since it's not explicitly laid out in the spec as far as I am aware). It seems like netlink is the only way to communicate with Wireguard in its kernelspace implementation.

I didn't look very closely into it, but I believe the library dependency I've added chooses automatically based on the conventions from the first-party userspace Wireguard implementation.